src/lib/update/clients/inst_update_partition_auto.rb

```diff
@@ -106,5 +108,20 @@ def target_system_candidate
 
       partitions.size == 1 ? target_root : nil
     end
+
+    # special version that respect online specific target distro
+    def target_distro
+      if Y2Packager::MediumType.online?
+        control_products = Y2Packager::ProductControlProduct.products
+        if control_products.empty?
```

Comment:
This whole `if` condition could be reduced to 

```ruby
# currently all products have the same "register_target" value
control_products.first&.register_target || ""
```

Reply:
hmm, I see that update does not use rubocop at all, so we can use new ruby features.